scripts/generate_prsn

```diff
@@ -0,0 +1,41 @@
+#!python
+from argparse import ArgumentParser
+import logging
+
+from dp.generate_prsn import logger, generate_prsn_file
+from nchelpers import CFDataset
+
+
+def main(args):
+    if args.metadata_check:
+        logger.info('Metadata Check')
+        for filepath in [args.prec, args.tasmin, args.tasmax]:
+            logger.info('')
+            logger.info('File: {}'.format(filepath))
+            try:
+                dataset = CFDataset(filepath)
+            except Exception as e:
+                logger.info('{}: {}'.format(e.__class__.__name__, e))
+            else:
+                for attr in 'project model institute experiment ensemble_member'.split():
```

Comment:
Maybe split this out into a utility function. It's a little dense for the `main()` function.